src/pages/services/CustumersService.ts

```diff
@@ -0,0 +1,56 @@
+import apiConfig from "@/pages/services/api.ts";
+import {DataResponseWrapper} from "@/pages/services/types/DataResponseWrapper.ts";
+import {CustomerWrapper, SearchCustomerQueryParamsType, SearchCustomerResponseType} from "@/pages/services/types/CustomerType.ts";
+
+export async function searchCustomer(searchQuery: SearchCustomerQueryParamsType): Promise<SearchCustomerResponseType> {
+    const response = await apiConfig
+        .get<DataResponseWrapper<SearchCustomerResponseType>>('/v1/customers', {
+            params: searchQuery
+        });
+
+    return response.data.data;
+}
+
+export async function findCustomer(customerId: string): Promise<CustomerWrapper> {
+    const response = await apiConfig
+        .get<DataResponseWrapper<CustomerWrapper>>(`/v1/customers/${customerId}`);
+
+    return response.data.data;
+}
+
+
+export async function createCustomer(customerData: Partial<CustomerWrapper>): Promise<CustomerWrapper> {
+    const response = await apiConfig
+        .post<DataResponseWrapper<CustomerWrapper>>('/v1/customers', customerData);
+
+    return response.data.data;
+}
+
+
+export async function updateCustomer(customerId: string, customerData: Partial<CustomerWrapper>): Promise<CustomerWrapper> {
+    const response = await apiConfig
+        .put<DataResponseWrapper<CustomerWrapper>>(`/v1/customers/${customerId}`, customerData);
+
+    return response.data.data;
+}
+
+
+export async function deleteCustomer(customerId: string): Promise<void> {
+    await apiConfig
+        .delete(`/v1/customers/${customerId}`);
+}
+
+
+export async function findCustomersAndNotifications(
+    customerId: string,
+    searchQuery: SearchCustomerQueryParamsType
+): Promise<SearchCustomerResponseType> {
+    const response = await apiConfig.get<DataResponseWrapper<SearchCustomerResponseType>>(
+        `/v1/customers/${customerId}/notifications`,
+        {
+            params: searchQuery
+        }
+    );
+
+    return response.data.data;
+}
```

Comment:
# Primeira sugestão
De acordo com wiki, o [Find Customer and Notifications](https://github.com/gina-system/gina-system-backend/wiki/Custumers-API#find-customer-and-notifications) só recebe como parâmetro o id do customers, acho q nao precisa passar o  `SearchCustomerQueryParamsType`, pode deixar só o customerId como param.

# Segundo ponto
Acredito que o retorno esteja errado, o retorno desse cara é igual ao do [Find Customer](https://github.com/gina-system/gina-system-backend/wiki/Custumers-API#find-customer-and-notifications), só que esse cara aqui tem uma propriedade a mais, tem uma lista de notificações junto. Esse type não ta criado ainda, vc pode ir no arquivo CustomerType e criar um type novo, tipo CustomerWithNotificationType q mapea todo CustomerType com a propriedade a mais. [Na wiki ](https://github.com/gina-system/gina-system-backend/wiki/Custumers-API#find-customer-and-notifications) ta listado em JSON como é o retorno certinho
ficando assim:
```typescript
export async function findCustomersAndNotifications(customerId: string): Promise<CustomerWithNotificationWrapper> {
    const response = await apiConfig
        .get<DataResponseWrapper<CustomerWithNotificationWrapper>>(`/v1/customers/${customerId}/notifications`);

    return response.data.data;
}

// O type ficaria assim:
export type CustomerWithNotificationWrapper = { customer: CustomerWithNotificationType };
```